ifttt/ifttt.php

```diff
@@ -1,181 +1,203 @@
 	$s .= '<h4>URL</h4>';
 	$s .= '<p>' . $a->get_baseurl() . '/ifttt/' . $a->user['nickname'] . '</p>';
 	$s .= '<h4>Method</h4>';
 	$s .= '<p>POST</p>';
 	$s .= '<h4>Content Type</h4>';
 	$s .= '<p>application/x-www-form-urlencoded</p>';
-	$s .= '<h4>'.t("Body for 'new status message'").'</h4>';
-	$s .= '<p><code>'.htmlentities('key='.$key.'&type=status&msg=<<<{{Message}}>>>&date=<<<{{UpdatedAt}}>>>&url=<<<{{PageUrl}}>>>').'</code></p>';
-	$s .= '<h4>'.t("Body for 'new photo upload'").'</h4>';
-	$s .= '<p><code>'.htmlentities('key='.$key.'&type=photo&link=<<<{{Link}}>>>&image=<<<{{ImageSource}}>>>&msg=<<<{{Caption}}>>>&date=<<<{{CreatedAt}}>>>&url=<<<{{PageUrl}}>>>').'</code></p>';
-	$s .= '<h4>'.t("Body for 'new link post'").'</h4>';
-	$s .= '<p><code>'.htmlentities('key='.$key.'&type=link&link=<<<{{Link}}>>>&title=<<<{{Title}}>>>&msg=<<<{{Message}}>>>&description=<<<{{Description}}>>>&date=<<<{{CreatedAt}}>>>&url=<<<{{PageUrl}}>>>').'</code></p>';
-        $s .= '</div><div class="clear"></div>';
-
-        $s .= '<div id="ifttt-rekey-wrapper">';
-        $s .= '<label id="ifttt-rekey-label" for="ifttt-checkbox">' . t('Generate new key') . '</label>';
-        $s .= '<input id="ifttt-checkbox" type="checkbox" name="ifttt-rekey" value="1" />';
-        $s .= '</div><div class="clear"></div>';
+	$s .= '<h4>' . t('Body for "new status message"') . '</h4>';
+	$s .= '<p><code>' . htmlentities('key=' . $key . '&type=status&msg=<<<{{Message}}>>>&date=<<<{{UpdatedAt}}>>>&url=<<<{{PageUrl}}>>>') . '</code></p>';
+	$s .= '<h4>' . t('Body for "new photo upload"') . '</h4>';
+	$s .= '<p><code>' . htmlentities('key=' . $key . '&type=photo&link=<<<{{Link}}>>>&image=<<<{{ImageSource}}>>>&msg=<<<{{Caption}}>>>&date=<<<{{CreatedAt}}>>>&url=<<<{{PageUrl}}>>>') . '</code></p>';
+	$s .= '<h4>' . t('Body for "new link post"') . '</h4>';
+	$s .= '<p><code>' . htmlentities('key=' . $key . '&type=link&link=<<<{{Link}}>>>&title=<<<{{Title}}>>>&msg=<<<{{Message}}>>>&description=<<<{{Description}}>>>&date=<<<{{CreatedAt}}>>>&url=<<<{{PageUrl}}>>>') . '</code></p>';
+	$s .= '</div><div class="clear"></div>';
+
+	$s .= '<div id="ifttt-rekey-wrapper">';
+	$s .= '<label id="ifttt-rekey-label" for="ifttt-checkbox">' . t('Generate new key') . '</label>';
+	$s .= '<input id="ifttt-checkbox" type="checkbox" name="ifttt-rekey" value="1" />';
+	$s .= '</div><div class="clear"></div>';
 
 	$s .= '<div class="settings-submit-wrapper" ><input type="submit" name="ifttt-submit" class="settings-submit" value="' . t('Save Settings') . '" /></div>';
-        $s .= '</div>';
-
+	$s .= '</div>';
 }
 
-function ifttt_settings_post(&$a,&$b) {
-
-        if(x($_POST,'ifttt-submit'))
-                if (isset($_POST['ifttt-rekey']))
-                        PConfig::delete(local_user(), 'ifttt', 'key');
+function ifttt_settings_post()
```

Comment:
I'm not sure if we musn't omit these parameters because they are sent with the hook..

Reply:
You can always pass more parameters to a function than it has declared, but not less, see `dba::p`. 😉